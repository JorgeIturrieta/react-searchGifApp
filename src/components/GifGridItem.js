import React from 'react' ;
import PropTypes from 'prop-types';

export const GifGridItem = ( {title,url}) => {

    return (
        <div className="card animate__animated animate__swing">
            <p> {title} </p>
            <img src={url} alt= {title}/>            
        </div>
    )       
}

GifGridItem.propTypes = {
 title: PropTypes.string.isRequired ,
 url: PropTypes.string.isRequired 
}
/* GifGridItem.defaultProps = {
    title: 'Default Title' ,
    url : 'https://localhost/algo.jpg' ,
}
 */