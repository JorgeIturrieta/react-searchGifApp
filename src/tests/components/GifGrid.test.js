import React from 'react';
import { GifGrid } from "../../components/GifGrid";
import { shallow } from "enzyme";
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid />', () => {
    const category = 'Dragon Ball' ;
    
    test('Debe de mostrar el componente correctamente ', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        }) ;
        const wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar items cuando se cargan imÃ¡genes con el custum hock useFetchGifs', () => {
        const gifs = [{
            id: '1' ,
            title: 'Anime 1' ,
            url: 'https://localhost/cualquier/cosa.jpg'
        },
        {
            id: '2' ,
            title: 'Anime 2' ,
            url: 'https://localhost/cualquier/cosa.jpg'
        }]
        useFetchGifs.mockReturnValue({

            data: gifs,
            loading: false,
        }) ;
        const wrapper = shallow(<GifGrid category={category} />)
       // expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);       
    })


})
