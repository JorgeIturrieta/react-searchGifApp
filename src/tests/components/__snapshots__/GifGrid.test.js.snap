// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe de mostrar el componente correctamente  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__tada"
  >
    Dragon Ball
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
